contracts/examples/erc1155-marketplace/src/lib.rs

```diff
@@ -0,0 +1,413 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+use elrond_wasm::HexCallDataSerializer;
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+const TRANSFER_TOKEN_ENDPOINT_NAME: &[u8] = b"safeTransferFrom";
+
+const PERCENTAGE_TOTAL: u8 = 100;
+
+#[derive(TopEncode, TopDecode, TypeAbi)]
+pub struct Auction<BigUint: BigUintApi> {
+	pub esdt_token: TokenIdentifier,
+	pub min_bid: BigUint,
+	pub max_bid: BigUint,
+	pub deadline: u64,
+	pub original_owner: Address,
+	pub current_bid: BigUint,
+	pub current_winner: Address,
+}
+
+#[derive(TopDecode, TypeAbi)]
+pub struct AuctionArgument<BigUint: BigUintApi> {
+	pub esdt_token: TokenIdentifier,
+	pub min_bid: BigUint,
+	pub max_bid: BigUint,
+	pub deadline: u64,
+}
+
+#[elrond_wasm_derive::contract(Erc1155MarketplaceImpl)]
+pub trait Erc1155Marketplace {
+	/// `bid_cut_percentage` is the cut that the contract takes from any sucessful bid
+	#[init]
+	fn init(&self, token_ownership_contract_address: Address, bid_cut_percentage: u8) {
+		self.set_token_ownership_contract_address(&token_ownership_contract_address);
+		self.set_percentage_cut(bid_cut_percentage);
+	}
+
+	// endpoints - Token ownership contract only
+
+	/// `_operator` argument is ignored, but it has to be kept because of the erc1155 standard
+	#[endpoint(onERC1155Received)]
+	fn on_erc1155_received(
+		&self,
+		_operator: Address,
+		from: Address,
+		type_id: BigUint,
+		token_id: BigUint,
+		args: AuctionArgument<BigUint>,
+	) -> SCResult<()> {
+		require!(
+			self.get_caller() == self.get_token_ownership_contract_address(),
+			"Only the token ownership contract may call this function"
+		);
+
+		sc_try!(self.try_create_auction(
+			&type_id,
+			&token_id,
+			&from,
+			&args.esdt_token,
```

Comment:
change name esdt_token to token_identifier ?

Reply:
renamed